invenio_vocabularies/datastreams/readers.py

```diff
@@ -159,6 +160,32 @@ def _iter(self, fp, *args, **kwargs):
             )
 
 
+class JsonLinesReader(BaseReader):
+    """JSON Lines reader."""
+
+    def _iter(self, fp, *args, **kwargs):
+        for idx, line in enumerate(fp):
+            try:
+                data = json.loads(line)
+                if isinstance(data, list):
+                    for entry in data:
+                        yield entry
+                else:
+                    yield data  # just one entry
+            except JSONDecodeError as err:
+                raise ReaderError(
+                    f"Cannot decode JSON line {fp.name}:{idx}: {str(err)}"
+                )
+
+
+class GzipReader(BaseReader):
+    """Gzip reader."""
+
+    def _iter(self, fp, *args, **kwargs):
```

Comment:
comment: I think this should be part of the `read`, otherwise the piping might not work. e.g.

1. Tar reader yields file paths
2. They are received by this reader, who's read function would `.open()` since it comes in the `item` parameter. Then it would try to `re-open` in the `_iter` steps... 

The same happened to the tar and zip reader whose functionality got splited.

p.s. sorry for eavesdropping again, just got a notification from GH 😅

Reply:
In the end, this one worked out since `gzip.open(...)` accepts file pointers as well (which is done in `BaseReader.read()`)